plugins/woocommerce/includes/class-wc-download-handler.php

```diff
@@ -580,13 +583,54 @@ private static function check_server_config() {
 	 * Can prevent errors, for example: transfer closed with 3 bytes remaining to read.
 	 */
 	private static function clean_buffers() {
-		if ( ob_get_level() ) {
-			$levels = ob_get_level();
-			for ( $i = 0; $i < $levels; $i++ ) {
-				@ob_end_clean(); // phpcs:ignore Generic.PHP.NoSilencedErrors.Discouraged
+		$levels = ob_get_level();
+		for ( $i = 0; $i < $levels; $i++ ) {
+			$ob_status = ob_get_status();
+
+			// can be not set in some installations
+			if ( ! isset( $ob_status['flags'] ) ) {
+				return;
 			}
-		} else {
-			@ob_end_clean(); // phpcs:ignore Generic.PHP.NoSilencedErrors.Discouraged
+
+			if ( ( $ob_status['flags'] & PHP_OUTPUT_HANDLER_REMOVABLE ) === PHP_OUTPUT_HANDLER_REMOVABLE ) {
```

Comment:
The fragment `$ob_status['flags'] & PHP_OUTPUT_HANDLER_REMOVABLE` seems potentially problematic under PHP 8.0 and higher, as you have an array on the left hand side of the `&`.

Reply:
@kkmuffme sorry for the delay here. I wanted to make time to test this properly, but on reflection I wonder if it would be better to first add tests that clearly show and validate how this will work with different layers of output buffering in place.